File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/NotifySubscriptionLapsedBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton

internal class NotifySubscriptionLapsedBridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {

    companion object {
        private const val notifySubscriptionLapsedErrorCode =
            "usercentrics_flutter_notifySubscriptionLapsed_error"
    }

    override val name: String
        get() = "notifySubscriptionLapsed"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)
        usercentrics.instance.notifySubscriptionLapsed(
            onSuccess = {
                result.success(null)
            },
            onError = {
                result.error(
                    notifySubscriptionLapsedErrorCode,
                    it.message,
                    it
                )
            },
        )
    }
}

File: ios/usercentrics_sdk/Sources/usercentrics_sdk/Bridge/NotifySubscriptionLapsedBridge.swift
import Flutter
import Foundation

struct NotifySubscriptionLapsedBridge : MethodBridge {

    let name: String = "notifySubscriptionLapsed"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)

        usercentrics.shared.notifySubscriptionLapsed(onSuccess: {
            result(nil)
        }, onError: { error in
            result(FlutterError(code: "usercentrics_flutter_notifySubscriptionLapsed_error",
                                message: error.localizedDescription,
                                details: nil))
        })
    }
}
